### ISSUE TITLE
import xmlschema fails, due to erroneous SCHEMAS_DIR url parsing

### ISSUE THREAD

**Issue by chrisjsewell** [brunato/xmlschema#116]
With python 3.6 on OSX, I get a `URLError` when trying to import `xmlschema`.

I've worked out, it is due to this line:
https://github.com/brunato/xmlschema/blob/481a4822e7ef26df17ebccadfebde7ae9f560d8a/xmlschema/validators/schema.py#L79
which sets `SCHEMAS_DIR` equal to e.g. `//anaconda/envs/myenv/lib/python3.6/site-packages/xmlschema/validators/schemas/`. Because of the initial double-forwardslash, calling
https://github.com/brunato/xmlschema/blob/481a4822e7ef26df17ebccadfebde7ae9f560d8a/xmlschema/resources.py#L83
then https://github.com/brunato/xmlschema/blob/481a4822e7ef26df17ebccadfebde7ae9f560d8a/xmlschema/resources.py#L95
creates an erroneous url, with 2 forward slashes instead of 3, i.e.

```python
import urllib
urlsplit = urllib.parse.urlsplit('//anaconda/envs/myenv/lib/python3.6/site-packages/xmlschema', scheme='file')
urlsplit.geturl()
'file://anaconda/envs/myenv/lib/python3.6/site-packages/xmlschema/'
```

```python
import urllib
urlsplit = urllib.parse.urlsplit('/anaconda/envs/myenv/lib/python3.6/site-packages/xmlschema', scheme='file')
urlsplit.geturl()
'file:///anaconda/envs/myenv/lib/python3.6/site-packages/xmlschema/'
```

If I remove the initial slash, everything works fine:

```python
SCHEMAS_DIR = os.path.join(os.path.dirname(__file__), 'schemas/').replace("//", "/")
```

**Comment by brunato** [comment]
Hi,
`os.path.join` and 'os.path.dirname` are library APIs, and the result of:

```python
os.path.join(os.path.dirname(__file__), 'schemas/')
```
seems to be incorrect. In this case this would not an issue of this library but of the python installation. 
Please check `os.path.dirname(__file__)` and then `os.path.join` individually to discover wich call returns an incorrect result. Maybe you can check also with other similar paths.

**Comment by chrisjsewell** [comment]
`__file__` returns `//anaconda/envs/...` (at least with python 3.6.5 on OSX)
As discussed [here](https://unix.stackexchange.com/questions/12283/unix-difference-between-path-starting-with-and/12285), a path starting with `//` is a legitimate posix path.
Any manipulation with `os` will handle it fine, but won't remove the `//`, i.e.

```python
>> os.path.exists("//anaconda/envs/myenv")
True
>> os.path.normpath("//anaconda/envs/myenv")
"//anaconda/envs/myenv"
```
It is the `urllib` module that handles these starting `//` differently, as discussed [here](https://stackoverflow.com/questions/8925938/url-parsing-in-python-normalizing-double-slash-in-paths)

>  In this case this would not an issue of this library but of the python installation

This may or may not be due to a bug in `urllib`, but if the library does not support an official python release, then really that comes down to an issue with the library. Actually, I'm surprised no one else has come across this bug.

Again, I'd reiterate that there is a very simple fix of:

```python
SCHEMAS_DIR = os.path.join(os.path.dirname(__file__), 'schemas/').replace("//", "/")
```

FYI, I also tried with a fresh python 3.7 installation `conda create -n py37 python=3.7 xmlschema`, and get the same issue, but there was not any issue with python 2.7.

**Comment by brunato** [comment]
Ok well, so it's only a variant of a correct os.path implementation.
About the fix, with the evidence that `SCHEMAS_DIR` is correct, I think it's better to put a fix into `XMLResource` because this class maybe used with other similar paths.

**Comment by brunato** [comment]
Should be fixed with release v1.0.13 (release v1.0.12 removed for a path normalization error on Windows after the changes made for this issue ...).
Check it on OSX and if it works close the issue.
Bye

**Comment by chrisjsewell** [comment]
@brunato yes that works great, thanks :)
